### DISCUSSION TITLE
Plot: how to make text and stroke scale with zoom?

### DISCUSSION THREAD

**Discussion by tgross35** [emilk/egui#3277] [Q&A] [answered]
Is there a way to make plot objects like text and stroke scale with the zoom rather than remain fixed?

For example, a simple stroked square with some small text:

<img width="499" alt="image" src="https://github.com/emilk/egui/assets/13724985/daf1bb64-a4a7-4387-9d29-bec3102c69e0">

Zooming way in, the square itself gets much larger but the stroke and text stay the same:

<img width="421" alt="image" src="https://github.com/emilk/egui/assets/13724985/2afc35d5-e6a3-48c6-a7c2-3811c6935449">

Can these be made to couple somehow?

Alternatively, is there a way to get the zoom/scale of a plot without calculating it manually from the bounds? If so, a scale for each object could just be calculated.

Also note that in the start/stop point in the bottom left corner, the stroke is not complete

**Comment by YgorSouza** [comment, marked as answer]
You could do something like this:

```rs
let scale = plot_ui.transform().rect_from_values(&[0.0, 0.0].into(), &[1.0, 1.0].into()).size();
let text_size = scale.max_elem().clamp(0.001, 250.0);
let stroke_width = scale.max_elem().clamp(0.001, 2500.0) * 0.1;
plot_ui.text(Text::new([0.0, 0.0].into(), RichText::new("Plot Text").size(text_size)));
plot_ui.polygon(Polygon::new(PlotPoints::new(vec![[-4.0, -1.5], [4.0, -1.5], [4.0, 1.5], [-4.0, 1.5]])).stroke(Stroke::new(stroke_width, Color32::BLUE)));
```

You have to limit the font size because egui will panic if it is too large (250 is already pushing it). And you have to decide what should happen if the X and Y scales are different, since you can't distort the elements along one dimension.

As for the stroke not joining at the ends, it's missing the "stroke-linejoin" and "stroke-linecap" properties from SVG that would fix that, so you can open a feature request about it.

![image](https://github.com/emilk/egui/assets/43298013/df95d493-bdfe-43fd-a36c-d3e8c42ae694)
